doc: revisions info adjustments (#10939)
nrfconnect/sdk-nrf: doc/nrf/developing/code_base.rst

.. _dm-revisions:

Versions and revisions
**********************

The |NCS| version numbers are assigned based on very specific criteria.
They follow the ``MAJOR.MINOR.PATCH`` pattern (also further called as ``X.Y.Z``), with possible additional postfixes depending on the release.

+-------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+
| Release type                  | Release critera                                                            | Comments                                                                                   |
+===============================+============================================================================+============================================================================================+

> | ``MAJOR``                     | Introduces a large number of substantial changes across the board          | Version number is increased rarely.                                                        |

Reviewer: ```suggestion
| ``MAJOR``                     | Introduces a large number of substantial changes across the board.          | Major version number is increased rarely and the release is accompanied by a migration guide.                                                        |
```

+-------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+
| ``MINOR``                     | Introduces a new functionality.                                            | Version number is increased every time a major release is cut.                             |
|                               |                                                                            | Minor releases are the default types of an |NCS| release.                                  |
|                               |                                                                            | They introduce new functionalities and may break APIs.                                     |
+-------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+

> | ``PATCH``                     | Introduces a minor or a bugfix release.                                    | Patch releases only address functional issues but do not introduce a new functionality.    |

Reviewer: ```suggestion
| ``PATCH``                     | Introduces minor changes to existing functionalities or bugfixes.                                    | Patch releases only address functional issues and do not introduce new functionalities.    |
```

Author: rephrased this part a bit
